### PR TITLE
fix: weird tab behaviour in RSVP and CFP sections

### LINKED ISSUE #481: [BUG] Weird tab behavior on the Dashboard RSVP tab
### Describe the bug

For reasons that aren't clear to me, the tabs on the Dashboard RSVP page appear out of thin air based on user action.

weird-tab-behavior

### To Reproduce

* open the dashbaord
* open a city meetup
* open the RSVP page from the left-hand sidebar
* you should only see two tabs - "Overview", "Web form"
* Click on one of the tabs and a new tab should magically appear - "Insights"

### PR DESCRIPTION
**Pull request by harshtandiya**
## What type of PR is this? (check all applicable)
- [x] 🐛Bug Fix
- [x] 🧑‍💻Refactor

## Description

- Fixed the weird tab behavior bug, due to which the tabs in RSVP & CFP sections in dashboard were hallucinating before rendering. See #481 

## Related Issues & Docs

closes #481

### CODE REVIEW

File: dashboard/src/pages/Home.vue [hunk 1]
```diff
@@ -10,13 +10,18 @@
...
 import { inject, watch, ref } from 'vue'
-import { usePageMeta } from 'frappe-ui'
+import { usePageMeta, createResource } from 'frappe-ui'
 import SideNavbar from '@/components/NewAppSidebar.vue'
 import HeaderWithNav from '@/components/HeaderWithNav.vue'
 
 const session = inject('$session')
 const showNav = ref(false)
 
+const nav_items = createResource({
```
Comment (NagariaHussain) [inline dashboard/src/pages/Home.vue]: No snake case here

File: dashboard/src/pages/EventRsvp.vue [hunk 2]
```diff
@@ -3,91 +3,97 @@
     <EventHeader
       class="px-4 py-8 md:p-8"
       :event="event"
-      :form_exists="rsvp_exists"
+      :form_exists="new Boolean(has_rsvp.data)"
```
Comment (NagariaHussain) [inline dashboard/src/pages/EventRsvp.vue]:
```suggestion
      :form_exists="Boolean(has_rsvp.data)"
```

File: dashboard/src/pages/EventCfp.vue [hunk 3]
```diff
@@ -28,66 +28,70 @@ const event = createDocumentResource({
...
+      return
+    }
+    tabs.options.push({
+      label: 'Web Form',
+      route: `/event/${route.params.id}/cfp/create`,
+    })
+  },
+})
+
+const event_cfp = createResource({
```
Comment (NagariaHussain) [inline dashboard/src/pages/EventCfp.vue]: Again, let's not mix cases and maintain consistency.

File: dashboard/src/pages/EventCfp.vue [hunk 4]
```diff
@@ -28,66 +28,70 @@ const event = createDocumentResource({
...
+        event: route.params.id,
       },
+    }
+  },
+  auto: true,
+  onSuccess(data) {
+    resetTabs()
+    if (data > 0) {
+      event_cfp.fetch()
+      tabs.options.push(...[
```
Comment (NagariaHussain) [inline dashboard/src/pages/EventCfp.vue:55]:
```suggestion
      tabs.options.push(...[
```

Why the spread?

**Review by NagariaHussain — CHANGES_REQUESTED** [review]

File: dashboard/src/pages/EventCfp.vue (hunk 4 above)
Comment (harshtandiya) [reply dashboard/src/pages/EventCfp.vue:55]: without the spread it only shows the `Overview` tab

**Review by NagariaHussain — APPROVED** [review]

**Pull request merged**